magicctapipe/scripts/lst1_magic/IT_container_data_MC_bash_scripts/setting_up_config_and_dir.py

```diff
@@ -0,0 +1,618 @@
+"""
+This script facilitates the usage of other two scripts
+of the MCP, i.e. "lst1_magic_mc_dl0_to_dl1.py" and
+"magic_calib_to_dl1.py". This script is more like a
+"manager" that organizes the analysis process by:
+1) Creating the necessary directories and subdirectories.
+2) Generatign all the bash script files that convert the
+MAGIC and MC files from DL0 to DL1.
+3) Launching these jobs in the IT container.
+
+Notice that in this stage we only use MAGIC + MC data.
+No LST data is used here.
+
+Standard usage:
+$ python setting_up_config_and_dir.py (-c config.yaml)
+
+If you want to run only the MAGIC or only the MC conversion,
+you can do as follows:
+
+Only MAGIC:
+$ python setting_up_config_and_dir.py --analysis-type onlyMAGIC (-c config.yaml)
+
+Only MC:
+$ python setting_up_config_and_dir.py --analysis-type onlyMC (-c config.yaml)
+
+"""
+
+import argparse
+import glob
+import logging
+import os
+from pathlib import Path
+
+import numpy as np
+import yaml
+
+__all__ = [
+    "nsb_avg",
+    "config_file_gen",
+    "lists_and_bash_generator",
+    "lists_and_bash_gen_MAGIC",
+    "directories_generator",
+]
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+
+def nsb_avg(source, config, LST_list):
+    """
+    This function evaluates the average of the NSB distribution per run
```

Comment:
per run? I think this should be the NSB distribution of the runs provided in the conifguration file, right?

Reply:
This is the average of the runs' NSB level (the first script evaluates one value for each run in the provided list; here we evaluates the average of these values). We can try to make the doc string a bit clearer (the average of the distribution in which each entry is a run)

Reply:
I think the misunderstanding is that maybe by "run" here you mean not the data run but the "current analysis"?

Reply:
What about 'this function evaluates the average of the NSB-level values that have been evaluated by the forst script (one value per run)' ? A bit more 'verbose' but maybe clearer

Reply:
Ok, I suggest:

```
Evaluate the average of the NSB levels that have been evaluated by LSTnsb_MC.py (one value per run).
```